service/history/ndc/workflow_state_replicator.go

```diff
@@ -976,6 +991,13 @@ func (r *WorkflowStateReplicatorImpl) applySnapshotWhenWorkflowNotExist(
 	if err != nil {
 		return err
 	}
+
+	if isStateBased {
+		for _, event := range events {
+			mutableState.AddReapplyCandidateEvent(event)
```

Comment:
also need to update the ReapplyEvents call in transaction_manager_new_workflow.go